tests/spec/core/linter-rules/check-period-in-p.js

```diff
@@ -0,0 +1,60 @@
+fdescribe("Core Linter Rule - 'check-period-in-p'", () => {
+  const ruleName = "check-period-in-p";
+  const config = { lint: { [ruleName]: true } };
+  let rule;
+  beforeAll(async () => {
+    rule = await new Promise(resolve => {
+      require([`core/linter-rules/${ruleName}`], ({ rule }) => resolve(rule));
+    });
+  });
+  it("checks p ending without a period", async () => {
+    const doc = document.implementation.createHTMLDocument("test doc");
+    doc.body.innerHTML = `
+        <section>
+          <p id="no-fullstop">Should pass</p>
+          <p>Should fail.</p>
+          <p id="no-fullstop-at-end">Should. pass</p>
+          <p>Should. Fail.</p>
+          <p id="unnecessary-tests-after-fullstop">Should Pass. Tests</p>
+        </section>
+    `;
+
+    const noFullStop = doc.getElementById("no-fullstop");
+    const noFullStopAtEnd = doc.getElementById("no-fullstop-at-end");
+    const unnecessaryTestsAfterFullStop = doc.getElementById("unnecessary-tests-after-fullstop");
+    
+    const results = await rule.lint(config, doc);
+    const [result] = results;
+    
+    expect(result.offendingElements.length).toEqual(2);
+    expect(result.offendingElements[0]).toEqual(noFullStop);
+    expect(result.offendingElements[1]).toEqual(noFullStopAtEnd);
+  });
+  it("checks error message for p ending without a period", async () => {
+    const doc = document.implementation.createHTMLDocument("test doc");
+    doc.body.innerHTML = `
+        <section>
+          <p id="no-fullstop">Should pass</p>
+          <p>Should fail.</p>
+          <p id="no-fullstop-at-end">Should. pass</p>
+          <p>Should. Fail.</p>
+          <p id="unnecessary-tests-after-fullstop">Should Pass. Tests</p>
+        </section>
+    `;
+  
+    const results = await rule.lint(config, doc);
+    const [result] = results;
```

Comment:
Something is not right here... each offending element should have a unique result.

Reply:
That unique result is contained in the result array which gets formed. On iterating through result, we can get teach of the results.